test/behaviour/reasoner/verification/SoundnessVerifier.java

```diff
@@ -101,14 +114,46 @@ private boolean canExplanationBeVerified(Explanation explanation) {
     private void verifyAnswerAndCollectExplanations(ConceptMap answer, Transaction tx) {
         verifyExplainableVars(answer);
         answer.explainables().iterator().forEachRemaining(explainable -> {
-            tx.query().explain(explainable.id()).forEachRemaining(explanation -> {
+            List<Explanation> explanations = tx.query().explain(explainable.id()).toList();
+            iterate(explanations).forEachRemaining(explanation -> {
```

Comment:
this can be shortened back to the old style of `tx.query().explain(...).forEachRemaining(...)`